src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -745,3 +745,34 @@ TEST_F(ControllerScriptEngineLegacyTest, screenWillSentRawDataIfConfigured) {
     ASSERT_ALL_EXPECTED_MSG();
 }
 #endif
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetUndefinedOnUnknownCharset) {
+    const auto result = evaluate("engine.convertCharset('NULL', 'Hello!')");
+    EXPECT_EQ(qjsvalue_cast<QByteArray>(result), QByteArray());
+}
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetCorrectValueWellKnown) {
+    const auto result = evaluate(
+            "engine.convertCharset(engine.WellKnownCharsets.Latin9, 'Hello!')");
+
+    // int repr for 0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x21
+    // aka ISO-8859-15 encoding for 'Hello!'
+    const char expected[] = {72, 101, 108, 108, 111, 33};
+    EXPECT_EQ(qjsvalue_cast<QByteArray>(result), QByteArray(expected, 6));
+}
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetCorrectValueStringCharset) {
+    const auto result = evaluate("`${engine.convertCharset('ISO-8859-15', 'Hello!')}`");
+    // int repr for 0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x21
+    // aka ISO-8859-15 encoding for 'Hello!'
+    const char expected[] = {72, 101, 108, 108, 111, 33};
```

Comment:
just put the hex in there if you prefer ;)
```suggestion
    ISO-8859-15 encoded 'Hello!'
    const char expected[] = {0x48, 0x65, 0x6c, 0x6c, 0x6f, 0x21};
```